builtin/logical/pki/path_resign_crls.go

```diff
@@ -0,0 +1,274 @@
+	nextUpdateParam     = "next_update"
+	crlsParam           = "crls"
+	formatParam         = "format"
+)
+
+func pathResignCrls(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "issuer/" + framework.GenericNameRegex(issuerRefParam) + "/resign-crls",
+		Fields: map[string]*framework.FieldSchema{
+			issuerRefParam: {
+				Type: framework.TypeString,
+				Description: `Reference to a existing issuer; either "default"
+for the configured default issuer, an identifier or the name assigned
+to the issuer.`,
+				Default: defaultRef,
+			},
+			crlNumberParam: {
+				Type:        framework.TypeInt,
+				Description: `The sequence number to be written within the CRL Number extension.`,
+			},
+			deltaCrlNumberParam: {
+				Type: framework.TypeInt,
+				Description: `Using a non-zero value specifies the base CRL revision number to encode within
+ a Delta CRL indicator extension, otherwise the extension will not be added.`,
+				Default: 0,
+			},
+			nextUpdateParam: {
+				Type: framework.TypeString,
+				Description: `The amount of time the generated CRL should be
+valid; defaults to 72 hours.`,
+				Default: defaultCrlConfig.Expiry,
+			},
+			crlsParam: {
+				Type:        framework.TypeStringSlice,
+				Description: `A list of PEM encoded CRLs to combine, originally signed by the requested issuer.`,
+			},
+			formatParam: {
+				Type: framework.TypeString,
+				Description: `The format of the combined CRL, can be "pem" or "der". If "der", the value will be
+base64 encoded. Defaults to "pem".`,
+				Default: "pem",
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathUpdateResignCrlsHandler,
+			},
+		},
+
+		HelpSynopsis: `Combine and sign with the provided issuer different CRLs `,
+		HelpDescription: `Provide two or more PEM encoded CRLs signed by the issuer, 
+normally from separate Vault clusters to be combined and signed.`,
+	}
+}
+
+func (b *backend) pathUpdateResignCrlsHandler(ctx context.Context, request *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	if b.useLegacyBundleCaStorage() {
+		return logical.ErrorResponse("This API cannot be used until the migration has completed"), nil
+	}
+
+	issuerRef := getIssuerRef(data)
+	crlNumber := data.Get(crlNumberParam).(int)
+	deltaCrlNumber := data.Get(deltaCrlNumberParam).(int)
+	nextUpdateStr := data.Get(nextUpdateParam).(string)
+	rawCrls := data.Get(crlsParam).([]string)
+
+	format, err := getCrlFormat(data.Get(formatParam).(string))
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	nextUpdateOffset, err := time.ParseDuration(nextUpdateStr)
+	if err != nil {
+		return logical.ErrorResponse("invalid value for %s: %v", nextUpdateParam, err), nil
+	}
+
+	if nextUpdateOffset <= 0 {
+		return logical.ErrorResponse("%s parameter must be greater than 0", nextUpdateParam), nil
+	}
+
+	if crlNumber < 0 {
```

Comment:
Now that I see it next to the code... if our intent is to use this mostly to sign internally created CRLs, what if we have `-1` be a sentinel to use the next internal CRL number on this cluster and bump it, like we do in the regular CRL building case?

That way we can default to `-1` and not have an explicit CRL number for the most part... Thoughts?

Reply:
CRL number itself is relatively meaningless, you're free to increment it how you want, skip things, &c as a server operator. We didn't even have it until recently...

Not all previous CRLs are available from the API anyways (so you'd have to _immediately_ fetch the new CRL, and you'd be racing against more revocations or the operator manually regenning again anyways).

I don't think a caching proxy can really rely on it strictly incrementing -- you wouldn't even _know_ about the new CRL number (as its not exposed anywhere via API) unless you serve it (which we wouldn't serve this new unified CRL)... ...so you'd only see it when Vault regenerates its internal CRLs and you'd see it skip one and think "oh, that's weird" and move on since this newer Vault-served CRL would have a higher number anyways.

Since its not used for Vault's internal CRLs, we wouldn't set the last complete CRL number field, so deltas would continue to referenced the existing Vault-built complete CRL... and any bespoke deltas would reference this custom bespoke CRL number...

I dunno, I think its fairly safe? I don't think its really a foot-gun, we'd still auto-increment the CRL number.

---

I think the real worry here would be that there's no "at least" behavior -- if two clusters' CRL Numbers get wildly out of sync due to downtime, you _could_ run into issues where you'd use cluster A's later CRLNumber with an earlier cluster B's lower CRLNumber automatically, which'd be bad I suppose... so you'd have to switch back to manual CRLNumbers.

Maybe that's enough to warrant not storing CRLNumbers then I guess, or if so, store it in a unified (cross-cluster) storage location...